pkg/scheduler/eventhandlers.go

```diff
@@ -171,6 +173,22 @@ func (sched *Scheduler) updatePodInSchedulingQueue(oldObj, newObj interface{}) {
 
 	logger.V(4).Info("Update event for unscheduled pod", "pod", klog.KObj(newPod))
 	sched.SchedulingQueue.Update(logger, oldPod, newPod)
+
+	isSchedulingQueueHintEnabled := utilfeature.DefaultFeatureGate.Enabled(features.SchedulerQueueingHints)
+
+	// TODO: remove IsPodUpdated when QHint hits GA?
+	if !isSchedulingQueueHintEnabled && !queue.IsPodUpdated(oldPod, newPod) {
+		return
+	}
+
+	// SchedulingQueue.Update() above triggers PodItselfUpdate for this updated Pod
+	// and hence we have to exclude the Pod from this requeueing process.
+	events := framework.PodSchedulingPropertiesChange(newPod, oldPod, false)
+	for _, evt := range events {
```

Comment:
We have to record event handling metrics appropriately.

See `updateNodeInCache`.
https://github.com/kubernetes/kubernetes/blob/facfe33a81f6b5af5d1e8166031472bb1d63c54f/pkg/scheduler/eventhandlers.go#L112